### PR TITLE
Add some Jsonnet programs that take a while to execute.

### PR DESCRIPTION
**Pull request by sparkprime**

### CODE REVIEW

File: perf_tests/large_string_join.jsonnet [hunk 1]
```diff
@@ -0,0 +1,5493 @@
+{
+    text: std.join(',', [
```
Comment (sbarzowski) [inline perf_tests/large_string_join.jsonnet]: Can we avoid having thousands of lines here? Navigating this is pretty annoying...

Comment (sparkprime) [reply perf_tests/large_string_join.jsonnet]: you mean do `std.join(',', std.makeArray(n, function() 'x'))` ?

sure although it'd be interesting to see how performance differs between the two.

Comment (sparkprime) [reply perf_tests/large_string_join.jsonnet]: you made some benchmarks as well, right?

I'd like to have realistic ones if possible.

Comment (sbarzowski) [reply perf_tests/large_string_join.jsonnet]: Yeah, something like that.

For truly realistic examples a corpus of real code would be the best, probably.

**Comment by sparkprime** [comment]
The one that golang used up all my RAM actually takes 30% of the RAM in C++ so could be explained by a constant factor on the golang side.  A constant factor of at least 2 is expected in garbage collectors, so it's not unexpected.

**Pull request merged**